### PR TITLE
Local rings

### PR DESCRIPTION
**Pull request by MatthiasHu**
This PR provides a definition of (commutative) local rings, some simple consequences and two alternative characterizations.

### CODE REVIEW

File: Cubical/Algebra/CommRing/LocalRing.agda [hunk 1]
```diff
@@ -0,0 +1,229 @@
+{-
+  Definition of local rings.
+-}
```
Comment (felixwellen) [inline Cubical/Algebra/CommRing/LocalRing.agda:3]: Maybe a comment like: "Note that local rings are more intricate constructively. See "A Course in Constructive Algebra" by Mines, Richman & Ruitenberg, p. 96 ff, where the two characterizations in this file are used as a definition of local ring, but without the requirement that 1 is different from 0."

**Review by felixwellen — CHANGES_REQUESTED** [review]
Awesome! Only one documentation request...

File: Cubical/Algebra/CommRing/LocalRing.agda (hunk 1 above)
Comment (MatthiasHu) [reply Cubical/Algebra/CommRing/LocalRing.agda:14]: Nice, thanks! I also added a link to [the nlab page](https://ncatlab.org/nlab/show/local+ring).

**Review by felixwellen — APPROVED** [review]
Thanks! Great stuff :-)

**Review by felixwellen — APPROVED** [review]
Thanks! Great stuff :-)

**Pull request merged**